test/unit/helpers.lua

```diff
@@ -468,7 +469,7 @@ else
       end
       return total_bytes_written
     end,
-    close = ffi.C.close,
+    close = iswin() and io.close or ffi.C.close,
```

Comment:
`io.close` is for file objects, error message for `io.close(0)` and `type(io.open(…))` suggests that it is a userdata with `FILE *`. Unless you have a solid idea on how you are going to open pipes it is too early to make this change, and creating process should probably go first.

I would also suggest to not add any win32 code here at all, add `elseif iswin()` clause with a whole table above.

Reply:
Do all of the units tests need `pipe`?

Reply:
I wanted to avoid a separate clause but it's not an option with incompatible APIs and headers for Windows. I'll work on it after fixing the setup for cl.exe and any commands that don't work on Windows.

Reply:
*Absolutely* all unit tests are run in a separate process created with `fork()` and communicate with parent via pipes. There is also code that ensures that running test which uses libnvim-test.so not inside a separate process is not possible.